payment_api/controller/sales/handler.go

```diff
@@ -0,0 +1,137 @@
+package sales
+
+import (
+	"encoding/json" // "encoding/json" // package to encode and decode the json into struct and vice versa
+	// "fmt"
+
+	"github.com/gorilla/mux" // used to get the params from the route
+
+	"log"
+	"net/http" // used to access the request and response object of the api
+
+	"github.com/jpgsaraceni/mvp-backend/payment_api/models"
+	sl_service "github.com/jpgsaraceni/mvp-backend/payment_api/services/sales"
+
+
+)
+
+func CreateSale(w http.ResponseWriter, r *http.Request) {
+	// set the header to content type x-www-form-urlencoded
+	// Allow all origin to handle cors issue
+	w.Header().Set("Context-Type", "application/x-www-form-urlencoded")
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Methods", "POST")
+	w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
+
+	// create an empty payment method of type models.PaymentMethod
+	var sale models.Sale
+
+	// decode the json request to paymentMethod
+	err := json.NewDecoder(r.Body).Decode(&sale)
+
+	if err != nil {
+		log.Fatalf("Unable to decode the request body %v", err)
```

Comment:
Se eu não to enganado, o Fatalf aqui vai parar a execução do programa. Acho que o melhor seria Printf mesmo.